script.js

```diff
@@ -1,22 +1,118 @@
-const courseContainer = document.querySelector(".courses-container");
 const searchButton = document.querySelector(".search-btn");
+const courseList = document.querySelectorAll(".course-bar li");
+const sectionContainer = document.querySelector(".course-container");
 let courses = [];
+[...courseList].forEach(function(link) {
```

Comment:
because querySelectorAll doesn't return array , i think it return nodeList which forEach isn't a function in that class ,
so i used that [... arr] expression to spread that array Like object to array

Reply:
we can iterate over  NodeList with forEach,  check this out https://developer.mozilla.org/en-US/docs/Web/API/NodeList